app/db/handlers.py

```diff
@@ -40,41 +36,41 @@ async def create(self, data: BaseModel) -> T:
 
         return obj
 
-    async def get(self, id: str | UUID) -> T:
-        try:
-            return await self.session.get_one(self.model_cls, id)
-        except NoResultFound as e:
-            raise NotFoundError(f"{self.model_cls.__name__} with ID {str(id)} wasn't found") from e
-
-    async def update(self, id: str | UUID, data: BaseModel) -> T:  # pragma: no cover
-        stmt = (
-            update(self.model_cls)
-            .where(self.model_cls.id == id)  # type: ignore[attr-defined, arg-type]
-            .values(**data.model_dump(exclude_unset=True))
-            .returning(self.model_cls)
-        )
-        try:
-            result = await self.session.exec(stmt)  # type: ignore[call-overload]
-            obj = result.scalars().one()
-            await self.session.commit()
-            return obj
-        except NoResultFound as e:
-            raise NotFoundError(f"{self.model_cls.__name__} with ID {str(id)} wasn't found") from e
+    async def get(self, id: UUID | str) -> ModelT:
+        result = await self.session.get(self.model_cls, id)
+        if result is None:
+            raise NotFoundError(f"{self.model_cls.__name__} with ID {str(id)} wasn't found")
+        return result
+
+    async def update(self, id: str | UUID, data: dict[str, Any]) -> ModelT:
+        # First fetch the object to ensure polymorphic loading
+        obj = await self.get(id)
+        # Update attributes
+        for key, value in data.items():
+            setattr(obj, key, value)
+        await self.session.commit()
+        await self.session.refresh(obj)
+        return obj
 
-    async def fetch_page(self, limit: int = 50, offset: int = 0) -> Sequence[T]:
-        results = await self.session.exec(select(self.model_cls).offset(offset).limit(limit))
-        return results.all()
+    async def fetch_page(self, limit: int = 50, offset: int = 0) -> Sequence[ModelT]:
+        results = await self.session.execute(select(self.model_cls).offset(offset).limit(limit))
```

Comment:
What if a negative value is passed in limit and/or offset? I know it's probably not the case,but 

`if limit < 0 or offset < 0:
            raise ValueError("Limit and offset must be non-negative integers.")
`
Also, there is no way to catch an exception that the query might raise

Reply:
I will do in my next task.